src/main/java/Duke.java

```diff
@@ -1,10 +1,147 @@
+import java.util.Scanner;
+
 public class Duke {
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+
+        boolean exitFlag = false;
+        int index = 0;
+        // Problem will come if there are more than 100 tasks
+        Task[] list = new Task[100];
+        printWelcomeMessage();
+        while (!exitFlag) {
+            // Read User input
+            Scanner in = new Scanner(System.in);
+            String line = in.nextLine();
+            // process user input and decide what operation to use
+            String operation = extractOperationType(line);
+            String printStatement = null;
+
+            switch (operation){
+                case "bye":
+                    exitFlag = true;
+                    printStatement = printByeMessage();
+                    break;
+
+                case "list":
+                    printStatement = printFullList(list, index);
+                    index--;
+                    break;
+
+                case "done":
+                    markTaskAsDone(list,line);
+                    index--;
+                    break;
+
+                case "todo":
+                    String toDoDescription = extractDescriptionFromString("todo",line);
+                    ToDo t = new ToDo(toDoDescription);
+                    list[index] = t;
+                    printStatement = printTaskDescription(index, list);
+                    break;
+
+                case "deadline":
+                    String deadlineDescription = extractDescriptionFromString("deadline", line);
+                    String date = extractDateFromString(line);
+                    Deadline d = new Deadline(deadlineDescription, date);
+                    list[index] = d;
+                    printStatement = printTaskDescription(index, list);
+                    break;
+
+                case "event":
+                    String eventDescription = extractDescriptionFromString("event", line);
+                    String eventDate = extractDateFromString(line);
+                    Event e = new Event(eventDescription, eventDate);
+                    list[index] = e;
+                    printStatement = printTaskDescription(index, list);
+                    break;
```

Comment:
Great use of switch statement! Really improves readability and produces a clean looking code